packages/lib/src/spatial-navigation/components/virtualizedList/helpers/computeTranslation.ts

```diff
@@ -1,50 +1,63 @@
 import { ScrollBehavior } from '../VirtualizedList';
+import { getSizeInPxFromOneItemToAnother } from './getSizeInPxFromOneItemToAnother';
 
-const computeStickToStartTranslation = ({
+const computeStickToStartTranslation = <T>({
   currentlyFocusedItemIndex,
   itemSizeInPx,
-  nbMaxOfItems,
-  numberOfItemsVisibleOnScreen,
+  data,
+  maxPossibleLeftAlignedIndex,
 }: {
   currentlyFocusedItemIndex: number;
-  itemSizeInPx: number;
-  nbMaxOfItems: number;
-  numberOfItemsVisibleOnScreen: number;
+  itemSizeInPx: number | ((item: T) => number);
+  data: T[];
+  maxPossibleLeftAlignedIndex: number;
 }) => {
-  const maxPossibleLeftAlignedIndex = Math.max(nbMaxOfItems - numberOfItemsVisibleOnScreen, 0);
-  const leftAlignedIndex = Math.min(currentlyFocusedItemIndex, maxPossibleLeftAlignedIndex);
-  const scrollOffset = leftAlignedIndex * itemSizeInPx;
+  const scrollOffset =
+    currentlyFocusedItemIndex < maxPossibleLeftAlignedIndex
+      ? getSizeInPxFromOneItemToAnother(data, itemSizeInPx, 0, currentlyFocusedItemIndex)
+      : getSizeInPxFromOneItemToAnother(data, itemSizeInPx, 0, maxPossibleLeftAlignedIndex);
   return -scrollOffset;
 };
 
-const computeStickToEndTranslation = ({
+const computeStickToEndTranslation = <T>({
   currentlyFocusedItemIndex,
   itemSizeInPx,
-  numberOfItemsVisibleOnScreen,
+  data,
+  listSizeInPx,
+  maxPossibleRightAlignedIndex,
 }: {
   currentlyFocusedItemIndex: number;
-  itemSizeInPx: number;
-  numberOfItemsVisibleOnScreen: number;
+  itemSizeInPx: number | ((item: T) => number);
+  data: T[];
+  listSizeInPx: number;
+  maxPossibleRightAlignedIndex: number;
 }) => {
-  const rightAlignedIndex = Math.max(
-    currentlyFocusedItemIndex - numberOfItemsVisibleOnScreen + 1,
-    0, // Equals to minPossibleLeftAlignedIndex
-  );
-  const scrollOffset = rightAlignedIndex * itemSizeInPx;
+  const scrollOffset =
+    currentlyFocusedItemIndex > maxPossibleRightAlignedIndex
+      ? getSizeInPxFromOneItemToAnother(data, itemSizeInPx, 0, currentlyFocusedItemIndex) +
```

Comment:
the expression is way too complex, there are nested ternaries etc. etc.
there are ways to simplify this, for example `if (currentlyFocusedItemIndex <= maxPossibleRightAlignedIndex) return 0`

Reply:
once you've done that you can extract the item size computation in a separate variable (that has a ternary indeed)

Reply:
Done :)